llvm/lib/Transforms/ObjCARC/ObjCARCOpts.cpp

```diff
@@ -2473,6 +2472,11 @@ bool ObjCARCOpt::run(Function &F, AAResults &AA) {
                             (1 << unsigned(ARCInstKind::AutoreleaseRV))))
     OptimizeReturns(F);
 
+  // Optimizations for autorelease pools.
+  if (UsedInThisFunction & ((1 << unsigned(ARCInstKind::AutoreleasepoolPush)) |
+                            (1 << unsigned(ARCInstKind::AutoreleasepoolPop))))
+    OptimizeAutoreleasePools(F);
```

Comment:
```suggestion
    Changed |= OptimizeAutoreleasePools(F);
```

I know the others don't do this, but these should probably return whether or not they changed the IR. As it stands, it's not clear to me that `Changed` will be guaranteed to be `true` if this optimization kicks in.